src/http-adapter.js

```diff
@@ -76,32 +80,39 @@ class HttpAdapter {
    * @returns {Promise} Promise for destroy.
    */
   destroy (link) {
-    return this._ajax('DELETE', link)
+    return this._http(METHOD.DELETE, link)
   }
 
-  _ajax (type, url, data) {
-    let headers = {
-      'Accept': 'application/vnd.api+json',
-      'Content-Type': 'application/vnd.api+json'
+  async _http (method = '', url = '', data = {}) {
+    let init = {
+      method,
+      headers: {
+        'Accept': 'application/vnd.api+json',
+        'Content-Type': 'application/vnd.api+json'
+      }
     }
 
-    // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
-    // some instrumented Vue data)
-    if (data && ['PATCH', 'POST'].indexOf(type) > -1) {
-      data = JSON.stringify(data)
+    url = new URL(url, location.origin)
+
+    if (method.includes(METHOD.GET)) {
```

Comment:
I'm curious as to why this isn't `method === METHOD.GET`? Similar for the later condition checks.

Reply:
Good point. It's inconsistent and I can clean that up